### PR TITLE
JENKINS-61985 Support 'myView' with configuration as code

### PR DESCRIPTION
**Pull request by timja**
See [JENKINS-61985](https://issues.jenkins-ci.org/browse/JENKINS-61985).

also https://github.com/jenkinsci/configuration-as-code-plugin/issues/1530  https://github.com/jenkinsci/configuration-as-code-plugin/issues/1837

This allows round tripping all default user properties from an initial setup.

### Proposed changelog entries

* Allow setting a user's primary view via configuration as code.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/model/MyViewsProperty.java
Comment (timja) [inline core/src/main/java/hudson/model/MyViewsProperty.java:143]: setUser is set via https://github.com/jenkinsci/jenkins/blob/e27b310065b3c036b5fc9d123f1d1d99d3058c00/core/src/main/java/hudson/model/User.java#L204

Which will be done later on not via jcasc

similar ish PR back in 2020 https://github.com/jenkinsci/jenkins/pull/4557

**Review by res0nance — APPROVED** [review]

**Comment by MarkEWaite** [comment]
I was unable to make a round-trip work for me with this change.  I merged a local copy of this to the master branch, built the jenkins.war, and ran it.  

I loaded the configuration as code and the folders plugins and created a view for my user, then generated the configuration as code definition and saved that as `jenkins.yaml`.  It included my custom view definition.

When I restarted Jenkins, my custom view definitions were gone but the user defined view was still correctly defined in my user configuration.  I think that means this proposed change worked, but the configuration round trip somehow did not include the custom view definition for my user, so it was lost on the configuration reload that happened at restart.

**Comment by timja** [comment]
ping @MarkEWaite

**Review by jetersen — APPROVED** [review]

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Comment by lkisac** [comment]
Hi @timja , I'm seeing this error on our Jenkins with latest LTS 2.387.2 and config-as-code plugin version is at 1625.v27444588c3d.  Prior version used was 1569.vb_fa_3896786a_7 and did not have this issue (this version I believe had this PR merged).  Shall I open a new ticket?

**Pull request merged**
